watertap/unit_models/osmotically_assisted_reverse_osmosis_0D.py

```diff
@@ -0,0 +1,212 @@
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#
+###############################################################################
+
+# Import Pyomo libraries
+from pyomo.environ import (
+    Var,
+    Set,
+    NonNegativeReals,
+    NegativeReals,
+    Reference,
+    units as pyunits,
+    exp,
+    value,
+    check_optimal_termination,
+)
+
+from idaes.core import declare_process_block_class, FlowDirection
+from idaes.core.util import scaling as iscale
+from idaes.core.util.misc import add_object_reference
+from watertap.core import (
+    MembraneChannel0DBlock,
+    ConcentrationPolarizationType,
+    MassTransferCoefficient,
+    PressureChangeType,
+)
+from watertap.core.membrane_channel0d import CONFIG_Template
+from watertap.unit_models.osmotically_assisted_reverse_osmosis_base import (
+    OsmoticallyAssistedReverseOsmosisBaseData,
+    _add_has_full_reporting,
+)
+import idaes.logger as idaeslog
+
+
+__author__ = "Adam Atia, Chenyu Wang"
+
+
+@declare_process_block_class("OsmoticallyAssistedReverseOsmosis0D")
+class OsmoticallyAssistedReverseOsmosisData(OsmoticallyAssistedReverseOsmosisBaseData):
+    """
+    Standard OARO Unit Model Class:
+    - zero dimensional model
+    - steady state only
+    - single liquid phase only
+    """
+
+    CONFIG = CONFIG_Template()
+
+    _add_has_full_reporting(CONFIG)
+    # TODO: add a test for this exception
+    def _add_membrane_channel_and_geometry(
+        self, side="feed_side", flow_direction=FlowDirection.forward
+    ):
+        if not isinstance(side, str):
+            raise TypeError(f"{side} is not a string. Please provide a string.")
+
+        # Build membrane channel control volume
+        setattr(
+            self,
+            side,
+            MembraneChannel0DBlock(
+                dynamic=False,
+                has_holdup=False,
+                property_package=self.config.property_package,
+                property_package_args=self.config.property_package_args,
+            ),
+        )
+        mem_side = getattr(self, side)
+
+        if (self.config.pressure_change_type != PressureChangeType.fixed_per_stage) or (
+            self.config.mass_transfer_coefficient == MassTransferCoefficient.calculated
+        ):
+            if not hasattr(self, "length") and not hasattr(self, "width"):
+                self._add_length_and_width()
+            mem_side.add_geometry(
+                length_var=self.length,
+                width_var=self.width,
+                flow_direction=flow_direction,
+            )
+            if not hasattr(self, "eq_area"):
+                add_eq_area = True
+            else:
+                add_eq_area = False
+            self._add_area(include_constraint=add_eq_area)
+        else:
+            mem_side.add_geometry(length_var=None, width_var=None)
```

Comment:
@adam-a-a is the issue with `MembraneChannel0DBlock` that the `flow_direction` is not passed in here?

Reply:
Seems like a good thing that we should add although not sure whether it'll serve as the remedy.